quisp/modules/QRSA/HardwareMonitor/HardwareMonitor.cc

```diff
@@ -177,25 +161,51 @@ void HardwareMonitor::handleMessage(cMessage *msg) {
   if (dynamic_cast<LinkTomographyResult *>(msg) != nullptr) {
     /*Link tomography measurement result/basis from neighbor received.*/
     LinkTomographyResult *result = check_and_cast<LinkTomographyResult *>(msg);
-
+    int partner_addr = result->getPartner_address();
     // Get QNIC info from neighbor address.
-    QNIC local_qnic = search_QNIC_from_Neighbor_QNode_address(result->getPartner_address());
-    auto it = all_temporal_tomography_output_holder[local_qnic.address].find(result->getCount_id());
-    if (it != all_temporal_tomography_output_holder[local_qnic.address].end()) {
-      EV << "Data already found.";
-      tomography_outcome temp = it->second;
-      if (result->getSrcAddr() == my_address) {
-        temp.my_basis = result->getBasis();
-        temp.my_output_is_plus = result->getOutput_is_plus();
-        temp.my_GOD_clean = result->getGOD_clean();
+    int qnic_addr_to_partner = routing_daemon->return_QNIC_address_to_destAddr(partner_addr);
+    auto local_qnic_info = findConnectionInfoByQnicAddr(qnic_addr_to_partner);
+    if (local_qnic_info == nullptr) {
+      error("local qnic info should not be null");
+    }
+    InterfaceInfo inter_info = getQnicInterfaceByQnicAddr(local_qnic_info->qnic.index, local_qnic_info->qnic.type);
+    QNIC local_qnic = inter_info.qnic;
+
+    // 1. find partner
+    auto partner_outputs_iter = temporal_tomography_output[local_qnic.address].find(partner_addr);
+    if (partner_outputs_iter != temporal_tomography_output[local_qnic.address].end()) {
+      // partner info found in this output
+      auto partner_output_iter = temporal_tomography_output[local_qnic.address][partner_addr].find(result->getCount_id());
+      if (partner_output_iter != temporal_tomography_output[local_qnic.address][partner_addr].end()) {
```

Comment:
you use `temporal_tomography_output[local_qnic.address][partner_addr]` three times in this part.
make a variable for this and assign it. it's too long and makes me tired to understand.

how about like this?
`auto partner_tomography_outputs = temporal_tomography_output[local_qnic.address][partner_addr]`;

Reply:
Right. Let me replace it with this.